### ISSUE TITLE
[🐞]: Wilder Wild Hollowed Logs Have odd Textures and BOP Palm Variant Missing

### ISSUE THREAD

**Issue by PoseironDire** [MehVahdJukaar/WoodGood#734]
### Code of Conduct

- [X] Always check you are using the latest version of the mods and its dependencies
- [X] Remove mod that enhances Minecraft: Optifine, Sodium, others. The issue still persists.
- [X] If you are unsure which mod is the culprit.
Disable all of your mods and enable them 1-2 mods each time to isolate the culprit
- [X] Confirm that there is no existing issue with a similar description submitted in the list of issues.

### Version - Loader (DO NOT FORGET TO SELECT THE CORRECT VERSION)

1.21.1 - FABRIC

### EveryCompat Version

everycomp-1.21-2.7.6-fabric

### Moonlight Lib Version

moonlight-1.21-2.17.12-fabric

### Issue with mods

BiomesOPlenty-fabric-1.21.1-21.1.0.7 and deeperdarker-fabric-1.21-1.3.2.1.

### Issue Detail

Hollowed Logs from WilderWild-3.0.11-mc1.21.1 have broken(ish) textures for the mods i listed above (perhaps for other mods too) and the Biomes O' Plenty Palm Hollowed log is missing. Also Every Compat 2.8+ does not have the hollowed logs at all
![2024-12-23_14 49 40](https://github.com/user-attachments/assets/224019ad-c530-4628-bc41-a17555cca6b6)
![2024-12-23_14 51 03](https://github.com/user-attachments/assets/d971c5fa-e769-4aa1-849d-4f08fa3573a7)

### OPTIONAL: Latest.log | Crash-report Attachment

No crashed, just unintentional game design.

### OPTIONAL: To Produce

Download fabric-api-0.110.0+1.21.1, BiomesOPlenty-fabric-1.21.1-21.1.0.7, GlitchCore-fabric-1.21.1-2.1.0.0, everycomp-1.21-2.7.6-fabric (for the broken textures) || everycomp-1.21-2.8.2-fabric (to see the Hollowed Logs missing), moonlight-1.21-2.17.12-fabric, TerraBlender-fabric-1.21.1-4.1.0.5, WilderWild-3.0.11-mc1.21.1 (All latest version as of posting this). Load into a creative world. Observe the Every Compat Tab...

**Comment by PoseironDire** [comment]
The Images were taken in my modpack, but i tried a modpack with only the mods i mentioned in the "To Produce" section and had the same effect.

**Comment by Xelbayria** [comment]
Alright, I need the ID of these hollow_log so i can take a close look

**Comment by PoseironDire** [comment]
Should I send those IDs? And if so, all of them? I'm pretty sure most logs are affected, even beyond the mods i used, since as i mentioned; 2.8.0+ versions of Every Compat doesn't even have those hollowed logs at all, and the palm log for BOP is completely missing. But if it helps i'll do what i can, just tell me more specifically what you need.

**Comment by Xelbayria** [comment]
you can provide the IDs of these affected logs. so i can take a look at what's the problem.

> Every Compat doesn't even have those hollowed logs at all

U meant they aren't generated at all? 

> the palm log for BOP is completely missing

sounds like it's not included to generate the hollow_log.

**Comment by PoseironDire** [comment]
Do you want the log IDs from Every Compat or the other mods affected? Not sure if this is what you were asking for: deeperdarker:echo_log biomesoplenty:jacaranda_log biomesoplenty:fir_log biomesoplenty:redwood_log biomesoplenty:maple_log biomesoplenty:pine_log biomesoplenty:hellbark_log biomesoplenty:empyreal_log biomesoplenty:dead_log biomesoplenty:mahogany_log biomesoplenty:willow_log biomesoplenty:umbran_log biomesoplenty:magic_log and lastly biomesoplenty:palm_log (their stripped variants too)

**Comment by Xelbayria** [comment]
I should be more clear.  What i am looking is the ID of generated Blocks from EC. 

like this: `everycomp:q/biomesoplenty/hollowed_fir_log` 

you can see the id from ☝🏻  will tell me which wood mods and furniture mod is Quark 

with that, i can look at the texture of the generated block to see what's wrong.

**Comment by FreyrDev** [comment]
Yeah I just discovered the missing hollow logs issue with everycomp 2.8.0+ which I think is the more serious problem here, the relevant lines in everycomp-entries.json (`"wilderwild.hollowed_log"` and `wilderwild.stripped_hollowed_log`) get removed completely when loading with 2.8.0+.

For reference when checking in everycomp 2.7.6, the only log I have loaded that has that texture problem is `everycomp:ww/blockus/hollowed_white_oak_log`.

**Comment by Xelbayria** [comment]
Let's not focus on the 2.7.6. it's an older version. what I want to focus is the v2.8.0+ 

because 2.8.0 has major changes in the code to support an addon which will be coming in the future.

**Comment by Xelbayria** [comment]
ok back to work on this. <strike>and discovered that it wasn't EveryComp's issue. 

WilderWild has supports for other wood mods. You need to report that to them. </strike> EDIT: ignore the strike-through. I checked and I think it only support the recipe stuff

One more thing, WildWilder support on EveryCompat were disabled. Will enable it in the next version. 

---

**EDIT:**

Hang on.

🤔 The only mod I know has Hollowed_log are Twilight Forest, you don't happened to have it installed? 

If so, then that's normal.

**Comment by Xelbayria** [comment]
FYI, I moved the WilderWild Module to "COMMON" because it can be supported on FORGE with Sinytra Connector. I forgot to "enable" it via COMMON.  the FABRIC was disabled and removed.

**Comment by Xelbayria** [comment]
## RE-ENABLED WilderWild Module in v2.8.3

---

Forgot to enable WilderWild Module after being moved to COMMON

**Comment by Xelbayria** [comment]
Closing this now. This is no longer relevant.

**Comment by FreyrDev** [comment]
This should probably be reopened. Now that the disappearing logs have been fixed, the texture issue that was originally reported is back in the latest version.

**Comment by Xelbayria** [comment]
> This should probably be reopened. Now that the disappearing logs have been fixed, the texture issue that was originally reported is back in the latest version.

If you are talking about the weird texture. that's because EveryCompat supported three mods: Wilder Wild, Quark & The Twilight Forest

they all have _hollow_log_

if that's not the same issue. you need to be more specific. with your message, i only can guess what was the problem.

**Comment by FreyrDev** [comment]
Ah fair I guess the problem isn't that obvious in the OP's screenshots. This is what I mean with `everycomp:ww/blockus/hollowed_white_oak_log`:

![2024-12-30_22 42 17](https://github.com/user-attachments/assets/541c229f-1137-4db5-b239-bd44edca0d3d)

It's using colours from the inside wood instead of just the bark (This is the regular hollow log, not the stripped version). OP was probably also pointing out that logs with a more unusual pattern – as seen on the left side of their first screenshot – end up using a texture for hollow logs that's pretty different from the base logs, though I'm not sure how fixable that is given my (limited) understanding of how everycompat works.

(Also I'll just note that I have neither Quark nor The Twilight Forest installed)

**Comment by Xelbayria** [comment]
oh that image helped a lot. I know what's the problem. <strike> Can you create a new issue with the information above. </strike>

EDIT: 

> It's using colours from the inside wood instead of just the bark 

That's exactly the problem. you are correct. its' an easy fix. 

EDIT 2: 

nvm. it's Finished.

**Comment by Xelbayria** [comment]
@FreyrDev 

can you check to see if `biomesplenty:palm` has stripped_log`. if not, then that would explain why 

> the Biomes O' Plenty Palm Hollowed log is missing

**Comment by Xelbayria** [comment]
## FIXED in v2.8.5

---

generating textures using the wrong side of the texture for _hollow_log_

**Comment by PoseironDire** [comment]
Sorry for going silent for so long. It seems that the texture issue is indeed not created by this mod but the Wilder Wild mod, I reported it there and mentioned this report in it. The recent updates have fixed the missing logs issue, though the BOP Palm variant is still missing

Edit: I noticed that even the vanilla wood variant's hollowed logs also have inconsistent textures when using a texture pack. I copied the textures from the texture pack and used them for the hollowed logs to fix that issue. I tried doing the same for the logs made by EveryCompat but I have no idea where to put those textures and if that's even possible.

**Comment by Xelbayria** [comment]
Alright

You can create a custom Resource Pack. then add the texture by the block ID. for example
`everycomp:ww/biomesoplenty/hollowed_fir_log` 

then you would put the textures in `everycomp:textures/block/ww/biomesoplenty/hollowed_fir_log.png` 

that will override the current texture generated by EC.

**Comment by PoseironDire** [comment]
Thanks! 
![2025-01-03_04 59 40](https://github.com/user-attachments/assets/d7a3ca60-20a7-4ad3-a0e5-0c55451e49d2)
![2025-01-03_05 02 43](https://github.com/user-attachments/assets/b3272024-1c39-4c15-91ce-51f910bcedc6)

Turns out the Hollowed Logs intentionally have custom textures so that they are distinguishable from regular logs, but i personally prefer the consistency anyways so thanks for the help!

**Comment by Xelbayria** [comment]
Yea, i saw the issue you created via Wilder Wild's github. 

Check if Palm has "stripped_log" block. that may explain why the block is missing if it don't have one. But if it has one, then i'll look into why hollow_palm_log is missing.

**Comment by PoseironDire** [comment]
Still missing (2.8.4) version of EveryCompat for 1.21.1
![2025-01-03_05 23 37](https://github.com/user-attachments/assets/c6c3865f-6994-48d5-a8b7-886ab583dbee)

**Comment by Xelbayria** [comment]
let me rephrase my question: does Biomes O' Plenty have stripped_log for palm????? that's the only one required to generate the hollow_log from EC

**Comment by PoseironDire** [comment]
Yes it has a stripped log variant, it can be seen in the image

**Comment by Xelbayria** [comment]
k will check
